javaslang/src/main/java/javaslang/collection/AbstractMap.java

```diff
@@ -41,6 +41,24 @@
 
     @SuppressWarnings("unchecked")
     @Override
+    public M put(Tuple2<? extends K, ? extends V> entry) {
+        Objects.requireNonNull(entry, "entry is null");
+        return (M) put(entry._1, entry._2);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public M removeAll(Iterable<? extends K> keys) {
```

Comment:
Thanks. I know - a lot of work with the repetitions in the collection classes o_O